lib/msf/core/db_manager/vuln.rb

```diff
@@ -158,7 +160,9 @@ def report_vuln(opts)
           sname = opts[:proto]
         end
 
-        service = host.services.where(port: opts[:port].to_i, proto: proto).first_or_create
+        services = host.services.where(port: opts[:port].to_i, proto: proto)
+        services = services.where(name: sname) if sname.present?
```

Comment:
We might need to change this again in the future; if multiple calls to this API have different `snames` present - odd behavior may occur

Reply:
Yeah, I'm actually wondering if this is closer to what we want:

```
Mdm::Host.first.services.where(port: 6379, proto: 'tcp').first_or_create(name: 'redis')
```

So we still want to support a generic lookup of not needing to provide sname, but still being able to assign a service name to a created service it's not already created.

Although that still doesn't handle the scenario of an existing service with a nil name being present, and a caller having a valid sname that it could be set to. So maybe there isn't a nice oner liner that would work here